frontend/catalyst/utils/jax_extras.py

```diff
@@ -994,3 +1018,125 @@ def bind(self, *args, **params):
         eqn = new_jaxpr_eqn(invars, outvars, self, params, [], source_info)
         trace.frame.add_eqn(eqn)
         return out_tracers if self.multiple_results else out_tracers.pop()
+
+
+def _gather_shape_rule_dynamic(
+    operand,
+    indices,
+    *,
+    dimension_numbers,
+    slice_sizes,
+    unique_indices,
+    indices_are_sorted,
+    mode,
+    fill_value,
+):  # pragma: no cover
+    """This is a modified version of the ``jax._src.lax.slicing._gather_shape_rule`` function. In
+    this version we skip the dimension check if the dimenstions are dynamic. The same fix is
+    suggested to the upstream as https://github.com/google/jax/pull/19083"""
```

Comment:
We should still include their license & copyright notice here, even if this function is copied temporarily. See example here: 
https://github.com/PennyLaneAI/catalyst/blob/5e177244e68b6eac3c4a23747ab433f9821c3134/frontend/catalyst/utils/jax_extras.py#L367-L371

Reply:
👍